pkg/pvc/pvc_protected_entity.go

```diff
@@ -283,6 +290,72 @@ func (this PVCProtectedEntity) getProtectedEntityForPV(ctx context.Context, pv *
 				return nil, errors.Errorf("Unexpected access mode, %v, for Persistent Volume %s", pv.Spec.AccessModes[0], pv.Name)
 			}
 		}
+	} else if pv.Spec.VsphereVolume != nil {
+		if pv.Spec.AccessModes[0] != core_v1.ReadWriteOnce {
+			return nil, errors.Errorf("Unexpected access mode, %v, for Persistent Volume %s", pv.Spec.AccessModes[0], pv.Name)
+		}
+		if pv.Annotations["pv.kubernetes.io/migrated-to"] == VSphereCSIProvisioner {
+			volumePath := pv.Spec.VsphereVolume.VolumePath
+
+			// Migrated volume must be of type ivd
+			pvPEType := this.getComponentPEType()
+
+			var config *rest.Config
+			path := os.Getenv("KUBECONFIG")
```

Comment:
It is not a good idea to get kubeconfig from env variable.

PVC PETM has a member variable of kubernetes client, which is constructed from `restConfig` parameter passed in when initializing the PVC PETM object. It might be better to create and expose a dynamic client there and use it here.

Reply:
Fixed

Reply:
changes in this file has not yet been synced?